server/protocols/jmap/src/test/java/org/apache/james/jmap/utils/JsoupHtmlTextExtractorTest.java

```diff
@@ -71,6 +71,64 @@ public void toPlainTextShouldUpperCaseLabels() {
     }
 
     @Test
+    public void toPlainTextShouldHandleListsWell() {
+        String html = "<ul>Here is my awesome list:" +
+            "  <li>JMAP</li>" +
+            "  <li>IMAP</li>" +
+            "</ul>" +
+            "<p>Followed with some text</p>" +
+            "<p>And some other text</p>";
+        String expectedPlainText = "Here is my awesome list:  \n" +
+            " - JMAP  \n" +
+            " - IMAP\n" +
+            "\n" +
+            "Followed with some text\n" +
+            "\n" +
+            "And some other text\n" +
+            "\n";
+        assertThat(textExtractor.toPlainText(html)).isEqualTo(expectedPlainText);
+    }
+
+    @Test
+    public void tableShouldBeWellHandled() {
+        String html = " <table style=\"width:100%\">\n" +
+            "  <tr>\n" +
```

Comment:
This Tika reverse engineering is going too far...

I think we should seriously wait for feedbacks before taking it any step furter...

Note that it's not that I don't know how to do this, but that I consider it harmful:
 - We don't have any idea of how far we should go... Only user feedbacks will tell us...
 - Handling of new elements adds new interactions in the display. As they can be mixed, for optimising display for specific HTML we create the risk of making it look worse for others...